### PR TITLE
Feat/add gridplus lattice1

### PR DESCRIPTION
**Pull request by mrcnk**
Fixes BX-####
Figma link (if any):

### CODE REVIEW

File: src/entries/popup/App.tsx [hunk 1]
```diff
@@ -33,6 +34,13 @@ import { RainbowConnector } from './wagmi/RainbowConnector';
 const playground = process.env.PLAYGROUND as 'default' | 'ds';
 const backgroundMessenger = initializeMessenger({ connect: 'background' });
 
+const getStoredClient = () => localStorage.getItem('storedClient') ?? '';
```
Comment (netbonus) [inline src/entries/popup/App.tsx]: lets put these functions in one spot and not duplicate them.

File: src/entries/popup/App.tsx [hunk 2]
```diff
@@ -75,6 +83,7 @@ export function App() {
       analytics.track(event.popupOpened);
       setTimeout(() => flushQueuedEvents(), 1000);
     }
+    setup({ getStoredClient, setStoredClient, name: 'Rainbow' });
```
Comment (netbonus) [inline src/entries/popup/App.tsx]: is there an easy way to check if they have an active gridplus account before running this function?

File: src/entries/popup/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -24,9 +26,12 @@ export function Checkbox({
   borderColorSelected?: BoxStyles['borderColor'];
   background?: 'accent' | BackgroundColor;
   backgroundSelected?: 'accent' | BackgroundColor;
+  testId?: string;
```
Comment (netbonus) [inline src/entries/popup/components/Checkbox/Checkbox.tsx]: Do we need these? it'd be nice not to have too many unrelated changes.

File: src/entries/popup/handlers/gridplus.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+      nonce: transaction.nonce
+        ? BigNumber.from(transaction.nonce).toNumber()
+        : undefined,
+      to: transaction.to || undefined,
+      value: transaction.value
+        ? BigNumber.from(transaction.value).toHexString()
+        : undefined,
+    };
+
+    let forceLegacy = false;
```
Comment (netbonus) [inline src/entries/popup/handlers/gridplus.ts]: `const forceLegacy = LEGACY_CHAINS_FOR_HW.includes(transaction.chainId as ChainId)`

File: src/entries/popup/handlers/gridplus.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    // HW doesn't support type 2 for these networks yet
+    if (LEGACY_CHAINS_FOR_HW.includes(transaction.chainId as ChainId)) {
+      forceLegacy = true;
+    }
+
+    if (transaction.gasPrice) {
+      baseTx.gasPrice = transaction.gasPrice;
+    } else if (!forceLegacy) {
+      baseTx.maxFeePerGas = transaction.maxFeePerGas || undefined;
```
Comment (netbonus) [inline src/entries/popup/handlers/gridplus.ts]: wont these be assigned to undefined if they don't exist?

File: src/entries/popup/handlers/gridplus.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+      });
+
+      const parsedTx = parse(serializedTransaction);
+      if (parsedTx.from?.toLowerCase() !== address?.toLowerCase()) {
+        throw new Error('Transaction was not signed by the right address');
+      }
+
+      return serializedTransaction;
+    } else {
+      console.log('gridplus error', JSON.stringify(response, null, 2), baseTx);
```
Comment (netbonus) [inline src/entries/popup/handlers/gridplus.ts]: does this match the logging and alerts across the app?

File: src/entries/popup/pages/hw/gridplus.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  step: GridplusStep;
+  setStep: (step: GridplusStep) => void;
+  onFinish: (addresses: string[]) => void;
+}) => {
+  switch (step) {
+    case GridplusStep.WALLET_CREDENTIALS:
+      return (
+        <WalletCredentials
+          appName="Rainbow"
+          onAfterSetup={(result) =>
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus.tsx:31]: would `onSetupSuccessful` be more descriptive?

File: src/entries/popup/pages/hw/gridplus.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            result
+              ? setStep(GridplusStep.ADDRESS_CHOICE)
+              : setStep(GridplusStep.PAIRING_SECRET)
+          }
+        />
+      );
+    case GridplusStep.PAIRING_SECRET:
+      return (
+        <PairingSecret
+          onAfterPair={() => setStep(GridplusStep.ADDRESS_CHOICE)}
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus.tsx:42]: same with `onPairingSuccessful`?

File: src/entries/popup/pages/hw/gridplus.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        />
+      );
+    case GridplusStep.ADDRESS_CHOICE:
+      return <AddressChoice onSelected={onFinish} />;
+    default:
+      return null;
+  }
+};
+
+export function ConnectGridPlus() {
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus.tsx:52]: we should follow the formatting conventions in the repo. They use spacing between code blocks for clarity.

File: src/entries/popup/pages/hw/gridplus/addressChoice.tsx [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+export type AddressChoiceProps = {
+  onSelected: (addressses: AddressesData['addresses']) => void;
+};
+
+export const AddressChoice = ({ onSelected }: AddressChoiceProps) => {
+  const { sortedAccounts } = useAccounts();
+  const [formData, setFormData] = useState({
+    selectedAddresses: [] as string[],
+  });
+  const [loadingAddresses, setLoadingAddresses] = useState(true);
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/addressChoice.tsx:30]: this codebase uses the convention of `isLoading` so like `isLoadingAddresses` and `setIsLoadingAddresses` is what they will expect. Same with `disabled` vs. `isDisabled`. 

We'll also want to break up the codeblocks with returns in here too.

File: src/entries/popup/pages/hw/gridplus/addressChoice.tsx [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+import { Checkbox } from '~/entries/popup/components/Checkbox/Checkbox';
+import { Spinner } from '~/entries/popup/components/Spinner/Spinner';
+import { useAccounts } from '~/entries/popup/hooks/useAccounts';
+
+export type AddressesData = {
+  addresses: Address[];
+};
+
+export type AddressChoiceProps = {
+  onSelected: (addressses: AddressesData['addresses']) => void;
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/addressChoice.tsx]:
```suggestion
  onSelected: (addresses: AddressesData['addresses']) => void;
```

File: src/entries/popup/pages/hw/gridplus/addressChoice.tsx [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+          (currentAddress) => currentAddress !== address,
+        ),
+      });
+    return setFormData({
+      selectedAddresses: [...formData.selectedAddresses, address],
+    });
+  };
+  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onSelected(formData.selectedAddresses as Address[]);
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/addressChoice.tsx]: why pass this event handler all the way down from `ConnectGridPlus`, when we could just do the thing right here?

File: src/entries/popup/pages/hw/gridplus/addressChoice.tsx [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+  };
+  useEffect(() => {
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/addressChoice.tsx:62]: this might be a bit clearer using `.then()` off of `fetchAddresses` so you can avoid having to `await` in the `useEffect`

File: src/entries/popup/pages/hw/gridplus/pairingSecret.tsx [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+import { i18n } from '~/core/languages';
+import { Box, Button, Text } from '~/design-system';
+import { Input } from '~/design-system/components/Input/Input';
+
+export type PairingSecretProps = {
+  onAfterPair?: () => void;
+};
+
+export const PairingSecret = ({ onAfterPair }: PairingSecretProps) => {
+  const [pairing, setPairing] = useState(false);
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/pairingSecret.tsx:15]: `isPairing, setIsPairing` is more clear

File: src/entries/popup/pages/hw/gridplus/walletCredentials.tsx [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+  };
+  useEffect(() => {
+    const checkPersistedClient = async () => {
+      if (getStoredClient()) {
+        const result = await setup({
+          getStoredClient,
+          setStoredClient,
+          name: appName,
+        });
+        onAfterSetup && onAfterSetup(result);
```
Comment (netbonus) [inline src/entries/popup/pages/hw/gridplus/walletCredentials.tsx:70]: we could use a context avoid a lot of the up/down message passing that is making this code so complicated.

**Review by netbonus — COMMENTED** [review]
looks really good. nice job

File: src/entries/popup/App.tsx (hunk 1 above)
Comment (mrcnk) [reply src/entries/popup/App.tsx]: Created reusable handlers for it

File: src/entries/popup/App.tsx (hunk 2 above)
Comment (mrcnk) [reply src/entries/popup/App.tsx]: Wrapped it with `getStoredGridPlusClient`

File: src/entries/popup/handlers/gridplus.ts (hunk 5 above)
Comment (mrcnk) [reply src/entries/popup/handlers/gridplus.ts]: Removed, it indeed resolves to `undefined`.

File: src/entries/popup/pages/hw/gridplus/walletCredentials.tsx (hunk 15 above)
Comment (mrcnk) [reply src/entries/popup/pages/hw/gridplus/walletCredentials.tsx:70]: Leaving it this way, but I cleaned the implementation a bit, so it's not that painful.

File: src/entries/popup/handlers/gridplus.ts (hunk 6 above)
Comment (mrcnk) [reply src/entries/popup/handlers/gridplus.ts]: Yes, the same way as Ledger and Trezor

File: src/entries/popup/pages/hw/gridplus.tsx (hunk 9 above)
Comment (mrcnk) [reply src/entries/popup/pages/hw/gridplus.tsx:52]: Automated with Prettier or didn't happen 😅

File: src/entries/popup/pages/hw/gridplus/addressChoice.tsx (hunk 10 above)
Comment (mrcnk) [reply src/entries/popup/pages/hw/gridplus/addressChoice.tsx:30]: Will clarify with Rainbow team when the actual PR is opened reg their preferences about this and spacing between blocks

**Review by netbonus — APPROVED** [review]
